### PR TITLE
Persisted Callout fixes

### PR DESCRIPTION
**Pull request by ddlbrena**
Persisted Callout fixes

This PR has two fixes:

1. Do not update callout content when the callout is hidden. The fix here is to add shouldComponentUpdate in CalloutContent.base and if the callout is hidden block the update.

2. Callout position is not properly updated. When testing things around I noticed that the callout position was not always properly updated. The fix is in componentWillUpdate to update the position when showing the callout.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8488)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Callout/CalloutContent.base.tsx [hunk 1]
```diff
@@ -122,8 +130,11 @@ export class CalloutContentBase extends BaseComponent<ICalloutProps, ICalloutSta
       this._setHeightOffsetEveryFrame();
     }
 
-    // if the callout becomes hidden, then remove any positions, bounds that were placed on it.
-    if (newProps.hidden && newProps.hidden !== this.props.hidden) {
+    // Ensure positioning is recalculated when we are about to show a persisted menu.
+    if (!newProps.hidden && newProps.hidden !== this.props.hidden) {
```
Comment (khmakoto) [inline packages/office-ui-fabric-react/src/components/Callout/CalloutContent.base.tsx]: Isn't `!newProps.hidden && newProps.hidden` always going to result in `false`?

Comment (ddlbrena) [reply packages/office-ui-fabric-react/src/components/Callout/CalloutContent.base.tsx]: !newProps.hidden is evaluated first, then newProps.hidden !== this.props.hidden, so this is, if the new value is not hidden, and the value is changing which in this case means from not hidden to hidden, run the code

Comment (joschect) [reply packages/office-ui-fabric-react/src/components/Callout/CalloutContent.base.tsx]: I think you should add another set of parenthesis to make that a bit more clear.

**Comment by khmakoto** [comment]
Closing and reopening to trigger screener status update.

**Pull request merged**